packages/core/helper-plugin/src/features/AppInfo.tsx

```diff
@@ -1,44 +1,35 @@
-import * as React from 'react';
+import { Context, ReactNode, createContext, useContext, useMemo } from 'react';
 
-import PropTypes from 'prop-types';
+type EmptyObject = Record<string, never>;
+
+type AppInfoContextValue = {
+  autoReload?: boolean;
+  communityEdition?: boolean;
+  currentEnvironment?: string;
+  dependencies?: Record<string, string>;
+  latestStrapiReleaseTag?: string;
+  nodeVersion?: string;
+  // projectId?: string;
+  setUserDisplayName: (name: string) => void;
+  shouldUpdateStrapi: boolean;
+  strapiVersion?: string;
+  useYarn?: boolean;
+  userDisplayName: string;
+  userId?: string;
+};
 
 /* -------------------------------------------------------------------------------------------------
  * Context
  * -----------------------------------------------------------------------------------------------*/
 
-/**
- * TODO: review this type, do we need to it all?
- */
-
-/**
- * @preserve
- * @typedef {Object} AppInfoContextValue
- * @property {boolean | undefined} autoReload
- * @property {boolean | undefined} communityEdition
- * @property {string | undefined} currentEnvironment
- * @property {Record<string, string>} dependencies
- * @property {string | null} latestStrapiReleaseTag
- * @property {string | undefined} nodeVersion
- * @property {string | undefined} projectId
- * @property {(name: string) => void} setUserDisplayName
- * @property {boolean} shouldUpdateStrapi
- * @property {string | undefined} strapiVersion
- * @property {boolean | undefined} useYarn
- * @property {string} userDisplayName
- * @property {string | null} userId
- *
- */
-
-/**
- * @preserve
- * @type {React.Context<AppInfoContextValue>}
- */
-const AppInfoContext = React.createContext();
+const AppInfoContext: Context<AppInfoContextValue | EmptyObject> = createContext({});
```

Comment:
```suggestion
const AppInfoContext = createContext<AppInfoContextValue | EmptyObject>({});
```

This is the better approach i think? I believe `createContext` has a generic...

Reply:
Oh yes thank you. I'll update the PR to include this.